apps/api/src/app/feature-flags/services/feature-flags.service.ts

```diff
@@ -48,4 +48,8 @@ export class FeatureFlagsService {
       }
     }
   }
+
+  public async get<T>(key: FeatureFlagKey, context: IFeatureFlagContext, defaultValue: T): Promise<T> {
+    return await this.service.get(key, context, defaultValue);
+  }
```

Comment:
You don't need to declare this function `async` because you do return `Promise`, unless you plan to have another code here, and you don't need to `await`.

Reply:
`this.featureFlagsService.get<string>(key, context, '');` this will pass the compiler because you don't care about the promise result, but the promise will run, even with your code (async/await) it will pass.
But this `const test = this.featureFlagsService.get<string>(key, context, '');` will assign the promise to the constant and then TS will tell you that. Anyway, from what I remember in the compilation time compiler can improve that code ;)